LmDataset.py

```diff
@@ -1378,10 +1378,20 @@ def init_seq_order(self, epoch=None, seq_list=None):
     else:
       num_seqs = self._get_data_len()
       self._seq_order = self.get_seq_order_for_epoch(
-        epoch=epoch, num_seqs=num_seqs, get_seq_len=lambda i: len(self._get_data(key=self.main_source_data_key, line_nr=i)))
+        epoch=epoch, num_seqs=num_seqs, get_seq_len=lambda i: self.get_seq_length(i)[self.main_source_data_key])
     self._num_seqs = len(self._seq_order)
     return True
 
+  def get_seq_length(self, sorted_seq_idx):
+    """
+    :type sorted_seq_idx: int
```

Comment:
Rather than using `type`, I think we should use `:param int sorted_seq_idx:`, which is more consistent to the remaining code, and more flexible.